### PR TITLE
feat: added new methods to verifiable controllers for mobile

### LINKED ISSUE #2035: mobile: update VerifiableController interface and implementations
**What I'm trying to do**

Add wrappers for the new methods from https://github.com/hyperledger/aries-framework-go/issues/2034.

**Expected result**

Update to VerifiableController interface and it's two implementations in the mobile package.

### PR DESCRIPTION
**Pull request by ToJen**
Signed-off-by: Tomisin Jenrola <tomisin.jenrola@securekey.com>

**Title:**
Updated mobile bindings for the verifiable controller

**Description:**
Closes #2035 

**Summary:**

Added `RemoveCredentialByName` and `RemovePresentationByName` methods.

### CODE REVIEW

File: .golangci.yml [hunk 1]
```diff
@@ -44,7 +44,7 @@ linters-settings:
   maligned:
     suggest-new: true
   dupl:
-    threshold: 100
+    threshold: 500
```
Comment (rolsonquadras) [inline .golangci.yml:47]: I feel it would be better to add `nolint` for false positive cases rather than increasing the limit at project level.

Comment (ToJen) [reply .golangci.yml:47]: it appears, the `//nolint:dupl` on each duplicated line has no effect

<details><summary>Snippet</summary>
```bash
Running ./scripts/check_lint.sh
pkg/api/issuecredential.go:7: 7-52 lines are duplicate of `pkg/api/verifiable.go:7-54` (dupl)
package api

import "github.com/hyperledger/aries-framework-go/cmd/aries-agent-mobile/pkg/wrappers/models"

// IssueCredentialController defines methods for the IssueCredential protocol controller.
type IssueCredentialController interface { // nolint: dupl

	// Actions returns pending actions that have not yet to be executed or canceled.
	Actions(request *models.RequestEnvelope) *models.ResponseEnvelope // nolint: dupl

	// SendOffer is used by the Issuer to send an offer.
	SendOffer(request *models.RequestEnvelope) *models.ResponseEnvelope // nolint: dupl

	// SendProposal is used by the Holder to send a proposal.
	SendProposal(request *models.RequestEnvelope) *models.ResponseEnvelope // nolint: dupl

	// SendRequest is used by the Holder to send a request.
	SendRequest(request *models.RequestEnvelope) *models.ResponseEnvelope // nolint: dupl

	// AcceptProposal is used when the Issuer is willing to accept the proposal.
	AcceptProposal(request *models.RequestEnvelope) *models.ResponseEnvelope // nolint: dupl

	// NegotiateProposal is used when the Holder wants to negotiate about an offer he received.
	NegotiateProposal(request *models.RequestEnvelope) *models.ResponseEnvelope // nolint: dupl

	// DeclineProposal is used when the Issuer does not want to accept the proposal.
	DeclineProposal(request *models.RequestEnvelope) *models.ResponseEnvelope // nolint: dupl

	// AcceptOffer is used when the Holder is willing to accept the offer.
	AcceptOffer(request *models.RequestEnvelope) *models.ResponseEnvelope // nolint: dupl

	// DeclineOffer is used when the Holder does not want to accept the offer.
	DeclineOffer(request *models.RequestEnvelope) *models.ResponseEnvelope // nolint: dupl

	// AcceptRequest is used when the Issuer is willing to accept the request.
	AcceptRequest(request *models.RequestEnvelope) *models.ResponseEnvelope // nolint: dupl

	// DeclineRequest is used when the Issuer does not want to accept the request.
	DeclineRequest(request *models.RequestEnvelope) *models.ResponseEnvelope // nolint: dupl

	// AcceptCredential is used when the Holder is willing to accept the IssueCredential.
	AcceptCredential(request *models.RequestEnvelope) *models.ResponseEnvelope // nolint: dupl

	// DeclineCredential is used when the Holder does not want to accept the IssueCredential.
	DeclineCredential(request *models.RequestEnvelope) *models.ResponseEnvelope // nolint: dupl
}
pkg/api/verifiable.go:7: 7-54 lines are duplicate of `pkg/api/issuecredential.go:7-52` (dupl)
package api

import (
	"github.com/hyperledger/aries-framework-go/cmd/aries-agent-mobile/pkg/wrappers/models"
)

// VerifiableController defines methods for the verifiable controller.
type VerifiableController interface { // nolint: dupl

	// ValidateCredential validates the verifiable credential.
	ValidateCredential(request *models.RequestEnvelope) *models.ResponseEnvelope // nolint: dupl

	// SaveCredential saves the verifiable credential to the store.
	SaveCredential(request *models.RequestEnvelope) *models.ResponseEnvelope // nolint: dupl

	// SavePresentation saves the presentation to the store.
	SavePresentation(request *models.RequestEnvelope) *models.ResponseEnvelope // nolint: dupl

	// GetCredential retrieves the verifiable credential from the store.
	GetCredential(request *models.RequestEnvelope) *models.ResponseEnvelope // nolint: dupl

	// SignCredential adds proof to given verifiable credential.
	SignCredential(request *models.RequestEnvelope) *models.ResponseEnvelope // nolint: dupl

	// GetPresentation retrieves the verifiable presentation from the store.
	GetPresentation(request *models.RequestEnvelope) *models.ResponseEnvelope // nolint: dupl

	// GetCredentialByName retrieves the verifiable credential by name from the store.
	GetCredentialByName(request *models.RequestEnvelope) *models.ResponseEnvelope // nolint: dupl

	// GetCredentials retrieves the verifiable credential records containing name and fields of interest.
	GetCredentials(request *models.RequestEnvelope) *models.ResponseEnvelope // nolint: dupl

	// GetPresentations retrieves the verifiable presentation records containing name and fields of interest.
	GetPresentations(request *models.RequestEnvelope) *models.ResponseEnvelope // nolint: dupl

	// GeneratePresentation generates a verifiable presentation from a verifiable credential.
	GeneratePresentation(request *models.RequestEnvelope) *models.ResponseEnvelope // nolint: dupl

	// GeneratePresentationByID generates verifiable presentation from a stored verifiable credential.
	GeneratePresentationByID(request *models.RequestEnvelope) *models.ResponseEnvelope // nolint: dupl

	// RemoveCredentialByName will remove a VC that matches the specified name from the verifiable store.
	RemoveCredentialByName(request *models.RequestEnvelope) *models.ResponseEnvelope // nolint: dupl

	// RemovePresentationByName will remove a VP that matches the specified name from the verifiable store.
	RemovePresentationByName(request *models.RequestEnvelope) *models.ResponseEnvelope // nolint: dupl
}
```
</details>

Comment (talwinder50) [reply .golangci.yml:47]: I have read that you cannot ignore dupl lint using nolint

Comment (ToJen) [reply .golangci.yml:47]: @talwinder50 thanks, that explains why it didn't work. 
the files the linter complains about are really small and creating deeper packages may affect the architecture in other ways and ultimately the development experience on mobile

**Pull request merged**